src/main/scala/org/camunda/feel/impl/interpreter/FeelInterpreter.scala

```diff
@@ -715,25 +724,37 @@ class FeelInterpreter {
     case _ => error(EvaluationFailureType.INVALID_TYPE, s"Can't divide '$x' by '$y'")
   }
 
-  private def unaryTestExpression(x: Val)(implicit context: EvalContext): Val =
+  private def unaryTestExpression(expression: Exp)(implicit context: EvalContext): Val = {
     withVal(
       input,
-      i =>
-        x match {
-          case ValBoolean(true)              => ValBoolean(true) // the expression is true
-          case ValList(ys) if ys.contains(i) =>
-            ValBoolean(true) // the expression contains the input value
-          case _ =>
-            checkEquality(i, x, _ == _, ValBoolean) match {
-              case ValBoolean(true)             => ValBoolean(true)  // the expression is the input value
-              case _ if x == ValBoolean(false)  => ValBoolean(false) // the expression is false
+      inputValue =>
+        eval(expression) match {
+          case _: ValFatalError                       =>
+            eval(expression)(context.add(INPUT_VALUE_SYMBOL -> inputValue)) match {
+              case ValBoolean(true)  => ValBoolean(true)
+              case ValBoolean(false) => ValBoolean(false)
+              case _                 => ValNull
+            }
```

Comment:
❓ I just want to verify I understand this correct:

1. We try to evaluate the expression with the original input value (null). Somehow this results in a fatal error.
2. We evaluate the expression again setting `?` to the input value (null) in the context. This will evaluate properly to:
    1. True if the expression evaluates to true (eg: `? = null`)
    2. False if the expression evaluates to false (eg `? != null`)
    3. Null if the expression doesn't return a boolean (eg `odd(?)`)


Before the evaluation would result in True because:

1. The fatal error didn't stop the pattern matching early. As a result we'd check equality: `checkEquality(i, x, _ == _, ValBoolean)`
2. `i` and `x` are both null, so they're equal resulting in True.

The entire concept of `?` didn't seem to exist before?

Reply:
Correct. :100: 

 > The entire concept of ? didn't seem to exist before?

Before, `?` was replaced with the input value. There was no check that the expression must return true. It was not strict enough.

Reply:
> Before, ? was replaced with the input value. There was no check that the expression must return true. It was not strict enough.

Thanks for the extra context! All good to merge imo 😄